packages/core/src/state-management/repository.ts

```diff
@@ -0,0 +1,85 @@
+import { Document } from '../document';
+import DocID from '@ceramicnetwork/docid';
+import { AsyncLruMap } from './async-lru-map';
+import { StateStore } from '../store/state-store';
+import { DocumentFactory } from './document-factory';
+import { DocState } from '@ceramicnetwork/common';
+
+export class Repository {
```

Comment:
This Repository is doing a lot less than I had imagined it would.  It seems like you've mostly turned the old `DocCache` into this new `Repository`, but they still do more or less the same thing.  I had expected the Repository to be an abstraction layer around the DocCache, StateStore, PinStore, and PinningBackends, but it seems like the latter 3 of those all still exist outside of the Repository, and the Repository has no knowledge of pinning.  Is the consolidation planned for a follow-up commit?  I'm worried about how many different but related objects we have that all have to do with managing document state in some form or fashion, and that there's a lot of cognitive overhead to developers to keep track of them all.